hoomd/md/PotentialPairGPU.cuh

```diff
@@ -421,6 +440,32 @@ struct PairForceComputeKernel
                                                                                compute_virial,
                                                                                tpp>);
 
+	    hipFuncAttributes attr;
+	    hipFuncGetAttributes(&attr, reinterpret_cast<const void*>(&gpu_compute_pair_forces_shared_kernel<evaluator, shift_mode, compute_virial, tpp>));
+            static unsigned int base_shared_bytes = UINT_MAX;
+            bool shared_bytes_changed = base_shared_bytes != shared_bytes + attr.sharedSizeBytes;
+            base_shared_bytes = static_cast<unsigned int>(shared_bytes + attr.sharedSizeBytes);
+
+            unsigned int max_extra_bytes
+                = static_cast<unsigned int>(pair_args.devprop.sharedMemPerBlock - base_shared_bytes);
+            static unsigned int extra_bytes = UINT_MAX;
+            if (extra_bytes == UINT_MAX || shared_bytes_changed)
+                {
+                // required for memory coherency
+                hipDeviceSynchronize();
+
+                // determine dynamically requested shared memory
+                char* ptr = static_cast<char*>(nullptr);
+                unsigned int available_bytes = max_extra_bytes;
+                for (unsigned int i = 0; i < typpair_idx.getNumElements(); ++i)
+                    {
+                    d_params[i].load_shared(ptr, available_bytes);
```

Comment:
This line is causing the code to crash, however all of the tests in `test_potential.py` pass when I comment out this line

Reply:
OK, I will take a look.

Reply:
Look at the way HPMC handles this. It has a separate `allocate_shared` call that doesn't attempt to write to the pointer. I don't know why the aniso potential doesn't, but it should.

Reply:
Did some quick digging here. I introduced the aniso code in #457, which was merged on Sept 16 2019. Jens introduce `allocate_shared` in #408, which was opened well before #457 but didn't get merged until about a month after. It looks like we just failed to synchronize those changes at the time since they appeared orthogonal.

Reply:
Looking further, the aniso potential is also using global arrays which can't hold nested data structures like managed arrays. I'm currently working to convert `PotentialPairGPU` to use `std::vector<param_type, managed_allocator<param_type> >` as HPMC does.

Reply:
Due to the size and long-lived nature of a lot of Jens's PRs that made these changes it's not surprising that we ended up out of sync, as I recall you and I were sort of feeling our way through Jens's unified memory handling when we worked on adding shared memory support to anisotropic potentials. Hopefully the changes aren't terribly difficult to make.